docs/general/naming.md

```diff
@@ -0,0 +1,312 @@
+- [Recommendations for Application Developers](#recommendations-for-application-developers)
+- [Attributes](#attributes)
+  - [otel.\* Namespace](#otel-namespace)
+  - [Attribute Name Pluralization Guidelines](#attribute-name-pluralization-guidelines)
+  - [Signal-specific Attributes](#signal-specific-attributes)
+- [Metrics](#metrics)
+  - [Naming Rules for Counters and UpDownCounters](#naming-rules-for-counters-and-updowncounters)
+    - [Pluralization](#pluralization)
+    - [Use `count` Instead of Pluralization for UpDownCounters](#use-count-instead-of-pluralization-for-updowncounters)
+    - [Do Not Use `total`](#do-not-use-total)
+  - [Instrument Naming](#instrument-naming)
+
+<!-- tocstop -->
+
+</details>
+
+## General Naming Considerations
+
+_This section applies to attribute names (also
+known as the "attribute keys"), as well as Metric and Event names. For brevity
+within this section when we use the term "name" without an adjective it is
+implied to mean either of these._
+
+Every name MUST be a valid Unicode sequence.
+
+_Note: we merely require that the names are represented as Unicode sequences.
+This specification does not define how exactly the Unicode sequences are
+encoded. The encoding can vary from one programming language to another and from
+one wire format to another. Use the idiomatic way to represent Unicode in the
+particular programming language or wire format._
+
+Names SHOULD follow these rules:
+
+- Names SHOULD be lowercase.
+
+- Use namespacing. Delimit the namespaces using a dot character.
+  For example `service.version` denotes the service version where
+  `service` is the namespace and `version` is an attribute in that namespace.
+
+- Namespaces can be nested. For example `telemetry.sdk` is a namespace inside
+  top-level `telemetry` namespace and `telemetry.sdk.name` is an attribute
+  inside `telemetry.sdk` namespace.
+
+  Use namespaces (and dot separator) whenever it makes sense. For example
+  when introducing an attribute representing a property of some object, follow
+  `*{object}.{property}` pattern. Avoid using underscore (`*{object}_{property}`)
+  if this object could have other properties.
+
+- For each multi-word dot-delimited component of the name separate the
+  words by underscores (i.e. use snake_case). For example
+  `http.response.status_code` denotes the status code in the http namespace.
+
+  Use underscore only when using dot (namespacing) does not make sense or changes
+  the semantical meaning of the name. For example, use `rate_limiting`
+  instead of `rate.limiting`.
+
+- Be precise. Attribute, event, metric, and other names should be descriptive and
+  unambiguous.
+  - When introducing a name describing certain property of the object,
+    include the property name. For example, use `file.owner.name` instead of `file.owner`
+    and `system.network.packets.dropped` instead of `system.network.dropped`
```

Comment:
@open-telemetry/semconv-system-approvers wdyt about this - should we rename `system.network.dropped` or change this sentence?

Reply:
created https://github.com/open-telemetry/semantic-conventions/issues/1700, let's follow up